src/katsdpcontroller/product_config.py

```diff
@@ -1396,6 +1397,81 @@ def from_config(
         )
 
 
+class GpucbfTiedArrayResampledVoltageStream:
+    """Real tied-array-resampled-voltage stream (GPU V-engine)."""
+
+    stream_type: ClassVar[str] = "gpucbf.tied_array_resampled_voltage"
+    _valid_src_types: ClassVar[_ValidTypes] = {"gpucbf.tied_array_channelised_voltage"}
+
+    def __init__(
+        self,
+        name: str,
+        src_streams: Sequence[Stream],
+        *,
+        n_chans: int,
+        pols: Sequence[str],
+        station_id: str,
+    ):
+        if len(src_streams) != 2:
+            raise ValueError(f"{self.stream_type} requires two {self._valid_src_types}")
+        grandparent_streams: List[GpucbfAntennaChannelisedVoltageStream] = [
+            tacv_src_streams.src_streams[0]
+            for tacv_src_streams in src_streams
+            if len(tacv_src_streams.src_streams) == 1
+            and isinstance(tacv_src_streams.src_streams[0], GpucbfAntennaChannelisedVoltageStream)
+        ]
+        _grandparent_stream_type = "gpucbf.antenna_channelised_voltage"
+        if len(grandparent_streams) != 2:
+            # Should be two in total, one per src_stream
+            raise ValueError(
+                f"{self.stream_type} requires {self._valid_src_types} with a single src_stream"
+            )
+        if grandparent_streams[0] != grandparent_streams[1]:
```

Comment:
```suggestion
        if grandparent_streams[0] is not grandparent_streams[1]:
```
The elements of `grandparent_streams` are Stream objects, not strings, and you want to check if they're the *same* object rather than just equivalent. It so happens that none of the types overload `__eq__` so equality is the same as object identity, but it's best not to rely on that.